packages/frontend/src/pages/NameFormPage.js

```diff
@@ -3,29 +3,42 @@ import { useParams, useNavigate } from "react-router-dom";
 import { FullWidthText } from "../components/Input/Text";
 import TextButtonInput from "../components/Input/TextButtonInput";
 import LoadingPage from "./LoadingPage";
+import { toast } from "react-hot-toast";
 
 export default function NameFormPage() {
 	const navigate = useNavigate();
 	const params = useParams();
 
 	const [loading, setLoading] = useState(true);
+
 	useEffect(() => {
-		// TODO: use context instead of localStorage
-		if (localStorage.getItem("chickName")) {
-			navigate(`/flock/${params.coopName}/lobby/`);
-		}
+		async function checkChickAndFlock() {
+			const chickId = localStorage.getItem("chickId");
+			if (chickId) {
+				const chickResponse = await fetch(
+					`${process.env.REACT_APP_API_URL}/flocks/${params.coopName}/chicks/${chickId}`
+				);
+				console.log(chickResponse.status);
+				if (chickResponse.status === 200) {
+					localStorage.setItem("chickName", chickResponse.name);
+					navigate(`/flock/${params.coopName}/lobby/`);
+					return;
+				}
+			}
+
+			localStorage.clear();
```

Comment:
does this clear all the flock data for that specific coopName?

Reply:
it just clears the browser's data - so it won't store old chickNames/Ids

Reply:
sounds good